pytorch_toolkit/formula_recognition/README.md

```diff
@@ -0,0 +1,159 @@
+# PyTorch realization of the Im2Markup
+
+This repository contains inference and training code for Im2LaTeX models.
+Source [repository](https://github.com/harvardnlp/im2markup/). This repository is a fork of [PyTorch realization](https://github.com/luopeixiang/im2latex/)
+Models code is designed to enable ONNX\* export and inference on CPU via OpenVINO™.
+
+## Setup
+
+### Prerequisites
+
+* Ubuntu\* 16.04
+* Python\* 3.7 or newer
+* PyTorch\* (1.4.0)
+* CUDA\* 10.1
+* OpenVINO™ 2020.1 with Python API
+
+### Installation
+
+Create and activate virtual environment:
+
+```bash
+bash init_venv.sh
+```
+
+
+### Download Datasets
+
+For training model One has to have dataset. Dataset format is similiar to [im2latex-100k](https://zenodo.org/record/56198#.X2NDQ2gzaUl). Main structure of the dataset is following:
+* `formulas_file` - file with one formula per line
+* `images_folder` - folder containing input images
+* `split_file` - the most important part of the dataset. This file contains `file_name` (tab symbol) `formula_idx` per line connecting corresponding index of the formula in the file with formulas and particular image with `image_name`. Example:
+    ```
+    11.png  11
+    34.png  34
+    ```
+    There should be at least two such files: `train_filter.lst` and `validate_filter.lst`
+
+> **NOTE**:
+> By default code asserts to see dataset structure with the following names:
+> `images_processed` - folder with images
+> `formulas.norm.lst` - file with preprocessed formulas, for details, refer to [im2markup](https://github.com/harvardnlp/im2markup) repository
+> `validate_filter.lst` and `train_filter.lst` - corresponding splits of the data.
+
+
+## Training
+
+To train Text Spotter model run:
+
+```bash
+python3 tools/train.py --config configs/train_config.yml --work_dir <path to work dir>
+```
+Work dir is used to store information about learning: saved model checkpoints, logs.
+
+### Description of the possible option in train config:
```

Comment:
of possible options in training configuration

Reply:
Fixed